src/go/rpk/pkg/cli/topic/analyze.go

```diff
@@ -0,0 +1,743 @@
+	percentileNames := []string{}
+	for _, p := range percentiles {
+		percentileNames = append(percentileNames, fmt.Sprintf("P%v", p))
+	}
+
+	const (
+		secSummary   = "summary"
+		secBatchRate = "partition batch rate (1/s)"
+		secBatchSize = "partition batch size (bytes)"
+	)
+
+	ts, err := cr.summarizeTopics(percentiles)
+	if err != nil {
+		return err
+	}
+	// Print the topic summaries in asc order by topic name
+	slices.SortFunc(ts, func(a topicSummary, b topicSummary) int {
+		return strings.Compare(a.Topic, b.Topic)
+	})
+
+	gs, err := cr.summarizeGlobal([]float64{50})
+	if err != nil {
+		return err
+	}
+
+	sections := out.NewMaybeHeaderSections(
+		out.ConditionalSectionHeaders(map[string]bool{
+			secSummary:   cfg.PrintGlobalSummary,
+			secBatchRate: cfg.PrintTopicSummary,
+			secBatchSize: cfg.PrintTopicSummary,
+		})...,
+	)
+
+	sections.Add(secSummary, func() {
+		tw := out.NewTabWriter()
+		tw.PrintColumn("topics", gs.Topics)
+		tw.PrintColumn("partitions", gs.Partitions)
+		tw.PrintColumn("total batch rate (1/s)", gs.TotalBatchesPerSecond)
+		tw.PrintColumn("p50 batch size (bytes)", gs.BatchSizes[0].value)
+		tw.Flush()
+	})
+
+	sections.Add(secBatchRate, func() {
+		tw := out.NewTable(append([]string{"TOPIC"}, percentileNames...)...)
+		defer tw.Flush()
+		for _, ts := range ts {
+			vals := []interface{}{}
+			for _, pv := range ts.BatchesPerSecond {
+				vals = append(vals, pv.value)
+			}
+
+			tw.Print(append([]interface{}{ts.Topic}, vals...)...)
+		}
+	})
+
+	sections.Add(secBatchSize, func() {
+		tw := out.NewTable(append([]string{"TOPIC"}, percentileNames...)...)
+		defer tw.Flush()
+		for _, ts := range ts {
+			vals := []interface{}{}
+			for _, pv := range ts.BatchSizes {
+				vals = append(vals, pv.value)
+			}
+
+			tw.Print(append([]interface{}{ts.Topic}, vals...)...)
+		}
+	})
+	return nil
+}
+
+type kvTPInfo struct {
+	TopicPartition topicPartition      `json:"topicPartition" yaml:"topic_partition"`
+	Info           *topicPartitionInfo `json:"info" yaml:"info"`
+}
+
+type rawResults struct {
+	TopicPartitionInfo []kvTPInfo `json:"topicPartitionInfo" yaml:"topic_partition_info"`
+	TimeRange          timeRange  `json:"timeRange" yaml:"time_range"`
+}
+
+func (cr collectResult) printRawResults(f config.OutFormatter, w io.Writer) (bool, error) {
+	// Json doesn't seem to like serializing the map types. So convert it to a slice here.
+	kvs := []kvTPInfo{}
+	for k, v := range cr.TopicPartitionInfo {
+		kvs = append(kvs, kvTPInfo{k, v})
+	}
+
+	res := rawResults{kvs, cr.TimeRange}
+	if isText, _, t, err := f.Format(res); !isText {
+		if err != nil {
+			return false, err
+		}
+
+		fmt.Fprintln(w, t)
+		return true, nil
+	}
+	return false, nil
+}
+
+const helpAnalyze = `Analyze topics.
+
+This command consumes records from the specified topics in order to determine
+characteristics of the topic like batch rate and batch size.
+
+Using the --format flag with either JSON or YAML prints all of the metadata collected.
```

Comment:
```suggestion
To print all of the metadata collected, use the --format flag with either JSON or YAML.
```
Just to clarify - if the --format flag isn't passed, does this only print a subset of the metadata, as opposed to all?

Reply:
Will make the requested change. And yeah, if the format flag isn't passed only a subset of the results in printed.